pgvectorscale/src/access_method/plain_storage.rs

```diff
@@ -213,22 +213,31 @@ impl<'a> Storage for PlainStorage<'a> {
     }
     fn get_full_distance_for_resort<S: StatsHeapNodeRead + StatsDistanceComparison>(
         &self,
+        scan: &PgBox<pgrx::pg_sys::IndexScanDescData>,
         qdm: &Self::QueryDistanceMeasure,
         _index_pointer: IndexPointer,
         heap_pointer: HeapPointer,
         meta_page: &MetaPage,
         stats: &mut S,
-    ) -> f32 {
+    ) -> Option<f32> {
         /* Plain storage only needs to resort when the index is using less dimensions than the underlying data. */
         assert!(meta_page.get_num_dimensions() > meta_page.get_num_dimensions_to_index());
 
-        let slot = unsafe { TableSlot::new(self.heap_rel, heap_pointer, stats) };
-        match qdm {
-            PlainDistanceMeasure::Full(query) => {
-                let datum = unsafe { slot.get_attribute(self.heap_attr).unwrap() };
-                let vec = unsafe { PgVector::from_datum(datum, meta_page, false, true) };
-                self.get_distance_function()(vec.to_full_slice(), query.to_full_slice())
-            }
+        let slot = unsafe {
+            TableSlot::from_index_heap_pointer(self.heap_rel, heap_pointer, scan.xs_snapshot, stats)
+        };
+        match slot {
+            None => None,
+            Some(slot) => match qdm {
+                PlainDistanceMeasure::Full(query) => {
+                    let datum = unsafe { slot.get_attribute(self.heap_attr).unwrap() };
+                    let vec = unsafe { PgVector::from_datum(datum, meta_page, false, true) };
+                    Some(self.get_distance_function()(
+                        vec.to_full_slice(),
+                        query.to_full_slice(),
+                    ))
+                }
+            },
```

Comment:
`?` on the Option might be nice here.
```suggestion
        let slot_opt = unsafe {
            TableSlot::from_index_heap_pointer(self.heap_rel, heap_pointer, scan.xs_snapshot, stats)
        };
        let slot = slot_opt?;
        match qdm {
            PlainDistanceMeasure::Full(query) => {
                let datum = unsafe { slot.get_attribute(self.heap_attr).unwrap() };
                let vec = unsafe { PgVector::from_datum(datum, meta_page, false, true) };
                Some(self.get_distance_function()(
                    vec.to_full_slice(),
                    query.to_full_slice(),
                ))
            }
        }
```

Reply:
But also, should we just propagate a None instead of the unwrap in `unsafe { slot.get_attribute(self.heap_attr).unwrap() };`? 

We could do `unsafe { slot.get_attribute(self.heap_attr)? };`.

Reply:
Ahh the ? is a good call. I thought there must be a better way. As for the get_attribute stuff -- I want this to error out (panic's result in Postgres errors, but I'll add a better message)